### PR TITLE
dts: dts-e2e.robot: add credentials test cases

### PR DESCRIPTION
**Pull request by DaniilKl**
Related to:
* https://github.com/Dasharo/dts-scripts/pull/93
* https://github.com/Dasharo/open-source-firmware-validation/issues/656

### CODE REVIEW

File: dts/dts-e2e.robot [hunk 1]
```diff
@@ -969,6 +969,14 @@ E2E006.004 Odroid H4 DPP update (Coreboot + UEFI -> Coreboot + UEFI) - with cred
     # 4) Start update:
     Go Through Update
 
+################################################################################
+# Credentials tests:
+################################################################################
+
+E2E007.001 QEMU Check credentials are being saved correctly
```
Comment (m-iwanicki) [inline dts/dts-e2e.robot]: Add test for empty password maybe if you are at it? As this is bug that had regression

Comment (DaniilKl) [reply dts/dts-e2e.robot]: Definitely, thank you

Comment (DaniilKl) [reply dts/dts-e2e.robot]: 64d53159bfa9b3473fec0e470746f3b3216fed1b

File: lib/dts-lib.robot [hunk 2]
```diff
@@ -412,3 +412,28 @@ Go Through Update Decline Heads
     Wait For Checkpoint And Write    ${DTS_HEADS_SWITCH_QUESTION}    N
     Wait For Checkpoint And Write    ${DTS_SPECIFICATION_WARN}    Y
     Wait For Checkpoint And Write    ${DTS_DEPLOY_WARN}    Y
+
+Check DPP Credentials In DTS
+    [Documentation]    This KW checks whether DTS saves provided credentials
+    ....    correctly.
+    [Arguments]    ${dpp_email}    ${dpp_password}
+    ${credentials}=    Execute Command In Terminal    cat /etc/cloud-pass
```
Comment (m-iwanicki) [inline lib/dts-lib.robot]: maybe you should also verify that credentials set in MinIO CLI (`mc`) changed?

Comment (DaniilKl) [reply lib/dts-lib.robot]: Good point

Comment (m-iwanicki) [reply lib/dts-lib.robot]: https://github.com/Dasharo/open-source-firmware-validation/blob/ceb08399cb28dfa9d1a8ac746a3703fb4852f4bd/dts/dts-e2e.robot#L77

**Comment by m-iwanicki** [comment]
```sh
robot -b cmd_logs.txt -v rte_ip:127.0.0.1 -v config:qemu \
            -L TRACE -v boot_dts_from_ipxe_shell:True \
            -v dts_ipxe_link:http://192.168.4.48:4321/dts.ipxe \
            -v snipeit:no -v dts_config_ref:refs/heads/main \
            -v dpp_email:"$CI_EMAIL" -v dpp_password:"$CI_PASSWORD" \
            -v dpp_email_fw_only:"$CI_DPP_EMAIL" -v dpp_password_fw_only:"$CI_DPP_PASSWORD" \
            -v dpp_email_extensions_only:"$CI_EXTENSIONS_EMAIL"  -v dpp_password_extensions_only:"$CI_EXTENSIONS_PASSWORD" \
            -v dpp_email_no_access:"$CI_NO_ACCESS_EMAIL" -v dpp_password_no_access:"$CI_NO_ACCESS_PASSWORD" \
            -t "E2E00[789]*" dts/dts-e2e.robot
```

Results:

<details><summary>Details</summary>
<p>

```text
==============================================================================
Dts-E2E
==============================================================================
E2E007.001 Check credentials are being saved correctly :: Check th... | PASS |
------------------------------------------------------------------------------
E2E007.002 Check old credentials are being overwritten by new :: M... | FAIL |
cloud-pass file should have only 2 lines: 4 != 2
------------------------------------------------------------------------------
E2E007.003 Check wrong credentials should not allow to log into DP... | FAIL |
'Dasharo DPP credentials have been saved
Cannot log in to DPP server.
Press Enter to continue' contains 'Dasharo DPP credentials have been saved'
------------------------------------------------------------------------------
E2E007.004 Check correct credentials should allow to log into DPP ... | PASS |
------------------------------------------------------------------------------
E2E007.005 Check empty e-mail should not pass :: Entering empty e-... | FAIL |
No match found for 'Press Enter to continue' in 15 seconds
Output:


Enter password:    .
------------------------------------------------------------------------------
E2E007.006 Check empty password should not pass :: Entering empty ... | FAIL |
No match found for 'Press Enter to continue' in 3 minutes
Output:

Dasharo DPP credentials have been saved
.
------------------------------------------------------------------------------
E2E007.008 Check DPP credentials with access to only firmware :: T... | PASS |
------------------------------------------------------------------------------
E2E007.009 Check DPP credentials with access to only extensions ::... | FAIL |
'Dasharo DPP credentials have been saved
Checking for Dasharo firmware...
Waiting for network connection ...
Network connection have been established!
Downloading board configs repository
Checking if board is Dasharo compatible.
Waiting for network connection ...
Network connection have been established!
A Dasharo Firmware binary has been found for your platform!
Your credentials give access to:
Dasharo Pro Package (DPP): YES
DTS Extensions: YES
Checking for available DTS extensions...
Installing available DTS extensions...
Installing package txeconfigtool-git-r0.core2_64.rpm...
Downloading package txeconfigtool-git-r0.core2_64.rpm...
`premium/des-packages/txeconfigtool-git-r0.core2_64.rpm` -> `/var/dasharo-package-manager/txeconfigtool-git-r0.core2_64.rpm`
Total: 0 B, Transferred: 21.64 KiB, Speed: 282.65 KiB/s
Package txeconfigtool-git-r0.core2_64.rpm have been downloaded successfully!
Package txeconfigtool-git-r0.core2_64 is already installed.
Dependencies resolved.
Nothing to do.
Complete!
Package txeconfigtool-git-r0.core2_64.rpm have been installed successfully!
Press Enter to continue' does not contain 'Dasharo Pro Package (DPP): NO'
------------------------------------------------------------------------------
E2E007.010 Check DPP credentials without DPP access :: Those crede... | PASS |
------------------------------------------------------------------------------
E2E007.011 Check DPP credentials with both DPP firmware and DTS ex... | PASS |
------------------------------------------------------------------------------
E2E008.001 Reboot UI Option Calls Reboot Command :: Reboot (R) UI ... | PASS |
------------------------------------------------------------------------------
E2E008.002 Poweroff UI Option Calls Poweroff Command :: Poweroff (... | PASS |
------------------------------------------------------------------------------
E2E008.003 Launch SSH Server UI Option Enables SSH Server Command ... | PASS |
------------------------------------------------------------------------------
E2E008.004 Enable Sending Logs UI Option Should Enable DTS Log Sen... | PASS |
------------------------------------------------------------------------------
E2E009.001 DTS extensions are installed and can be used :: Test th... | PASS |
------------------------------------------------------------------------------
Dts-E2E                                                               | FAIL |
15 tests, 10 passed, 5 failed
==============================================================================
```

</p>
</details> 

Failing tests are to be fixed in `dts-scripts`

File: dts/dts-e2e.robot [hunk 3]
```diff
@@ -53,6 +53,265 @@ E2E001.001 HCL Report test
...
+    ...    services and shouldn't be saved
+    Clean Up DTS Environment
+    Execute Command In Terminal    export DTS_TESTING="true"
+    Write Into Terminal    dts-boot
+
+    VAR    ${DPP_EMAIL}=    test@email.com    scope=TEST
+    VAR    ${DPP_PASSWORD}=    test-password    scope=TEST
+    ${out}=    Provide DPP Credentials
+    Should Contain    ${out}    Cannot log in to DPP server.
+    Should Not Contain    ${out}    Dasharo DPP credentials have been saved
```
Comment (aronowski) [inline dts/dts-e2e.robot:140]: Given that we first save the credentials, and later on perform the check, whether they can be used to log in, I wonder if it might be worthwhile to have a clearing mechanism instead, with a message like "Cannot log in to DPP server. The provided credentials have been cleared. Try again.".

Comment (m-iwanicki) [reply dts/dts-e2e.robot:140]: I think we should first check if MinIO accepts new credentials if it does then we can save them to `/etc/cloud-pass`.
Depending on how we will do it it'll change `Edit DPP keys` behaviour (whether it'll keep old, working, credentials or clear them all).

Comment (DaniilKl) [reply dts/dts-e2e.robot:140]: @m-iwanicki, @aronowski, you can create an issue in github.com/dasharo/dasharo-issues, so we could consider adding this to the upcoming releases.

Comment (aronowski) [reply dts/dts-e2e.robot:140]: dts-scripts logic to be fixed in https://github.com/dasharo/dasharo-issues/issues/1589

**Review by aronowski — APPROVED** [review]
